### PR TITLE
fix(plasma-web): prevent default safari datepicker appearance

### PR DESCRIPTION
**Pull request by Yakutoc**
В примере для storybook необходимо показать как можно скрыть нативный date picker для браузера Safari. 

Было: 

Снимок экрана 2023-01-23 в 13 23 47

Стало:

Снимок экрана 2023-01-23 в 13 23 14

:baby_chick: Download canary assets:

[colors--canary.287.3993143077.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.287.3993143077.xml)  
[color_metro_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.287.3993143077.swift)  
[color_metro_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.287.3993143077.kt)  
[color_metro_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.287.3993143077.ts)  
[color_sbermarket_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.287.3993143077.swift)  
[color_sbermarket_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.287.3993143077.kt)  
[color_sbermarket_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.287.3993143077.ts)  
[color_sberprime_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.287.3993143077.swift)  
[color_sberprime_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.287.3993143077.kt)  
[color_sberprime_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.287.3993143077.ts)  
[color_selgros_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.287.3993143077.swift)  
[color_selgros_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.287.3993143077.kt)  
[color_selgros_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.287.3993143077.ts)  
[color_smbusiness_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.287.3993143077.swift)  
[color_smbusiness_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.287.3993143077.kt)  
[color_smbusiness_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.287.3993143077.ts)  
[PlasmaTokensColor--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.287.3993143077.swift)  
[typo_mage_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.287.3993143077.swift)  
[typo_mage_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.287.3993143077.kt)  
[typo_mage_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.287.3993143077.ts)  
[typo_plasma_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.287.3993143077.swift)  
[typo_plasma_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.287.3993143077.kt)  
[typo_plasma_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.287.3993143077.ts)  
[typo_ruler_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.287.3993143077.swift)  
[typo_ruler_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.287.3993143077.kt)  
[typo_ruler_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.287.3993143077.ts)  
[typo_sage_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.287.3993143077.swift)  
[typo_sage_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.287.3993143077.kt)  
[typo_sage_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.287.3993143077.ts)  
[typo_sbermarket_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.287.3993143077.swift)  
[typo_sbermarket_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.287.3993143077.kt)  
[typo_sbermarket_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.287.3993143077.ts)  
[typo_soulmate_ios-swift--canary.287.3993143077.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.287.3993143077.swift)  
[typo_soulmate_kotlin--canary.287.3993143077.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.287.3993143077.kt)  
[typo_soulmate_react-native--canary.287.3993143077.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.287.3993143077.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.126.1-canary.287.3993143077.0
  npm install @salutejs/plasma-temple@1.123.1-canary.287.3993143077.0
  npm install @salutejs/plasma-ui@1.156.1-canary.287.3993143077.0
  npm install @salutejs/plasma-web@1.153.1-canary.287.3993143077.0
  # or 
  yarn add @salutejs/plasma-b2c@1.126.1-canary.287.3993143077.0
  yarn add @salutejs/plasma-temple@1.123.1-canary.287.3993143077.0
  yarn add @salutejs/plasma-ui@1.156.1-canary.287.3993143077.0
  yarn add @salutejs/plasma-web@1.153.1-canary.287.3993143077.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-287/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-287/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-287/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-287/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-287/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-287/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-287/temple-storybook/

**Comment by Yeti-or** [comment]
[https://user-images.githubusercontent.com/1813468/213992728-d6f94016-1287-4168-963f-434a08fcc1d9.MOV]()

На iOS похоже что то идёт все ещё не так

File: packages/plasma-web/src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -7,7 +7,9 @@ import type { InputProps } from '../Input';
 
 export interface TextFieldProps
     extends Pick<FieldProps, 'contentLeft' | 'contentRight' | 'helperText'>,
-        Omit<InputProps, 'hasContentLeft' | 'hasContentRight'> {}
+        Omit<InputProps, 'hasContentLeft' | 'hasContentRight'> {
+    onClick?: <T>(event: MouseEvent<T>) => void;
```
Comment (Yeti-or) [inline packages/plasma-web/src/components/TextField/TextField.tsx:11]: в TextFieldProps нет onClick ?

Comment (Yakutoc) [reply packages/plasma-web/src/components/TextField/TextField.tsx]: > в TextFieldProps нет onClick ?

Сам удивился такому но ts ругается если не объявить **явно** в интерфейсе. 

> Примечание: Добавил тип для onClick, потому что ругался TS  Property 'onClick' does not exist on type

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-287/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-287/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🔴 Fail**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-287/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-287/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-287/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-287/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-287/temple-storybook/

File: packages/plasma-web/src/components/TextField/TextField.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-web/src/components/TextField/TextField.tsx]: @Yeti-or Сделал `npx lerna bootstrap`

Все равно IDEA TS показывает.

Снимок экрана 2023-01-24 в 11 15 41

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-287/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-287/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🔴 Fail**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-287/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-287/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-287/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-287/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-287/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-web/src/components/TextField/TextField.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-web/src/components/TextField/TextField.tsx]: возможно это из-за разных версий ts, посмотри какой редактор что подсасывает

Comment (Yeti-or) [reply packages/plasma-web/src/components/TextField/TextField.tsx]: но вообще то что у нас стори в  exclude тоже странно

File: packages/plasma-web/src/components/Calendar/Calendar.stories.tsx [hunk 2]
```diff
@@ -300,12 +300,23 @@ export const WithPopup: Story<CalendarProps> = ({ min, max, isDouble }) => {
...
+
     return (
         <Popup
             isOpen={isOpen}
             trigger="click"
             placement="bottom"
-            disclosure={<StyledTextField type="date" value={textValue} onChange={handleOnTextChange} />}
+            disclosure={
+                <StyledTextField
+                    type="date"
```
Comment (neretin-trike) [inline packages/plasma-web/src/components/Calendar/Calendar.stories.tsx:313]: насколько я помню, мы тут хотели поставить type="text"

Comment (Yakutoc) [reply packages/plasma-web/src/components/Calendar/Calendar.stories.tsx:313]: @neretin-trike Мы говорили об этом в контексте, если не получиться для iOS обойти проблему показа нативного календаря. Но решение найдено, поэтому не стал усложнять.

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**